### PR TITLE
[#391] Added preface chapters

### PR DESCRIPTION
**Pull request by beikov**

### CODE REVIEW

File: documentation/src/main/asciidoc/entity-view/manual/en_US/01_general.adoc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+JPA applications that use the entity model in every layer often suffer from the infamous `LazyInitializationException` or _N + 1_ queries issues.
+This is mainly due to the use of a too general model for a use case and is often solved by making use of a specialized DTO and adapting queries to that structure.
+The use of DTOs normally requires adapting many parts of an application and a lot of boilerplate code which is why people tend to do _the wrong thing_ like making use of
+the _open session in view_ anti-pattern. Apart from lazy loading issues, also the performance suffers due to selecting _unnecessary data_ that a UI is never displaying.
+
+{projectname} entity views tries to solve these and many more problems a developer faces when having to implement efficient model mapping in a JPA application.
+It allows to define DTOs as interfaces and provide the mappings to the JPA model via annotations. It has smart defaults(convention-over-configuration) allowing to omit most mappings.
+The DTOs are applied to a query builder through the `ObjectBuilder` extension point making it possible to separate querying logic from the projections but still have high performance queries.
```
Comment (Mobe91) [inline documentation/src/main/asciidoc/entity-view/manual/en_US/01_general.adoc]: Better:
`By applying DTOs to a query builder through the `ObjectBuilder` extension point it is possible to separate query logic from the projections while still enjoying high performance queries.`

File: documentation/src/main/asciidoc/entity-view/manual/en_US/01_general.adoc [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+[preface]
+== System requirements
+
+{projectname} entity view requires at least Java 1.7 and at least a JPA 2.0 implementation.
```
Comment (Mobe91) [inline documentation/src/main/asciidoc/entity-view/manual/en_US/01_general.adoc]: To be consistent:
`{projectname} entity views require at least Java 1.7 and at least a JPA 2.0 implementation.`

File: documentation/src/main/asciidoc/entity-view/manual/en_US/01_general.adoc [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+It depends on the core module and requires the use of the same versions for both modules.
```
Comment (Mobe91) [inline documentation/src/main/asciidoc/entity-view/manual/en_US/01_general.adoc]: `The entity view module depends...`

File: documentation/src/main/asciidoc/core/manual/en_US/01_general.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
+:numbered!:
+
+[preface]
+== Preface
+
+Every JPA application that requires some kind of dynamic queries for e.g. filtering will have to decide between duplicating parts of queries or building queries conditionally.
```
Comment (Mobe91) [inline documentation/src/main/asciidoc/core/manual/en_US/01_general.adoc]: Try to stick to present form in the whole paragraph.

File: documentation/src/main/asciidoc/core/manual/en_US/01_general.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+JPA offers the Criteria API for constructing such dynamic queries, but using this API will often result in unreadable and hard to maintain code.
+Concatenating query string parts is often an alternative that might even work for simple cases, but will quickly fall apart in many real world scenarios.
+Implementing pagination with JPA and especially when fetching collections is hard to do efficiently and often sub-optimal ways are chosen for keeping maintainability.
+
+{projectname} is a library that lives on top of a JPA provider and tries to solve these and many more problems a developer faces when having complex requirements.
+It is composed of multiple modules that all depend on the core module which this documentation deals with.
+
+The core module tries to ease the pain of writing dynamic queries by offering a fluent builder API that puts readability first!
```
Comment (Mobe91) [inline documentation/src/main/asciidoc/core/manual/en_US/01_general.adoc]: I would not use an exclamation mark here. Just a dot would be better.

**Comment by beikov** [comment]
Merged by 1d870d5aa884acb56e8cec3e5f70b9c3069fb605

**Pull request closed without merging**